spec/safe_statements_spec.rb

```diff
@@ -2110,6 +2110,555 @@ def up
+                  safe_partman_create_parent :foos3, key: nil, interval: "monthly"
+                end
+              end
+
+              expect do
+                migration.suppress_messages { migration.migrate(:up) }
+              end.to raise_error(ArgumentError, "Expected <key> to be present")
+            end
+
+            it "raises error when interval not present" do
+              migration = Class.new(migration_klass) do
+                def up
+                  safe_partman_create_parent :foos3, key: :created_at, interval: nil
+                end
+              end
+
+              expect do
+                migration.suppress_messages { migration.migrate(:up) }
+              end.to raise_error(ArgumentError, "Expected <interval> to be present")
+            end
+
+            it "raises error when unsupported optional arg is supplied" do
+              migration = Class.new(migration_klass) do
+                def up
+                  safe_partman_create_parent :foos3, key: :created_at, interval: "monthly", foo: "bar"
+                end
+              end
+
+              expect do
+                migration.suppress_messages { migration.migrate(:up) }
+              end.to raise_error(ArgumentError, "Unrecognized optional argument(s): [:foo]")
+            end
+
+            it "raises error when on Postgres < 10" do
+              migration = Class.new(migration_klass) do
+                def up
+                  safe_partman_create_parent :foos3, key: :created_at, interval: "monthly"
+                end
+              end
+
+              allow(ActiveRecord::Base.connection).to receive(:postgresql_version).and_return(9_06_00)
+
+              expect do
+                migration.suppress_messages { migration.migrate(:up) }
+              end.to raise_error(PgHaMigrations::InvalidMigrationError, "Native partitioning not supported on Postgres databases before version 10")
+            end
+
+            it "raises error when parent table does not exist" do
+              migration = Class.new(migration_klass) do
+                def up
+                  safe_partman_create_parent :foos3, key: :created_at, interval: "monthly"
+                end
+              end
+
+              expect do
+                migration.suppress_messages { migration.migrate(:up) }
+              end.to raise_error(PgHaMigrations::InvalidMigrationError, "Could not find foos3 in search path")
+            end
+
+            it "raises error when parent table does not exist and fully qualified name provided" do
+              migration = Class.new(migration_klass) do
+                def up
+                  safe_partman_create_parent "public.foos3", key: :created_at, interval: "monthly"
+                end
+              end
+
+              expect do
+                migration.suppress_messages { migration.migrate(:up) }
+              end.to raise_error(ActiveRecord::StatementInvalid, /Please create parent table first: public.foos3/)
```

Comment:
In a handful of places, we're simply bubbling up an error that comes from partman. Do we want to capture these and reraise as an `PgHaMigrations::InvalidMigrationError`?

Reply:
I'd probably prefer validation first.

Reply:
It is a little difficult though, because partman does _a ton_ of validation, so we'd essentially be rewriting a chunk of partman in ruby. That being said though, we _do_ have (and need) some validation when doing table discovery. If you simply provide a table name that doesn't exist, you'll get a pg ha migrations error. If you provide a fully qualified table name that doesn't exist, you'll get statement invalid error from partman. So it's probably at least worth making those types of errors consistent. I'll look into that

Reply:
I did fix this _particular_ error so that we're now raising a `PgHaMigrations::InvalidMigrationError` for consistency